packages/firestore/test/unit/specs/spec_test_components.ts

```diff
@@ -90,12 +96,16 @@ export class MockIndexedDbPersistence extends IndexedDbPersistence {
     ) => PersistencePromise<T>
   ): Promise<T> {
     if (this.injectFailures) {
-      return Promise.reject(
-        new IndexedDbTransactionError(new Error('Simulated retryable error'))
-      );
-    } else {
-      return super.runTransaction(action, mode, transactionOperation);
+      if (this.injectFailures[action as PersistenceAction] === undefined) {
```

Comment:
If this block of code is an identical copy/paste from above, then consider moving it to a helper function.

Reply:
It's not identical because I only updated one of the two blocks to use "else if". But it should be identical... Moved it to a helper function.